sycl/include/CL/sycl/libcxx_span.hpp

```diff
@@ -0,0 +1,578 @@
+    template <class Container>
+        constexpr explicit(Extent != dynamic_extent) span(Container& cont);
+    template <class Container>
+        constexpr explicit(Extent != dynamic_extent) span(const Container& cont);
+    constexpr span(const span& other) noexcept = default;
+    template <class OtherElementType, size_t OtherExtent>
+        constexpr explicit(Extent != dynamic_extent) span(const span<OtherElementType, OtherExtent>& s) noexcept;
+    ~span() noexcept = default;
+    constexpr span& operator=(const span& other) noexcept = default;
+
+    // [span.sub], span subviews
+    template <size_t Count>
+        constexpr span<element_type, Count> first() const;
+    template <size_t Count>
+        constexpr span<element_type, Count> last() const;
+    template <size_t Offset, size_t Count = dynamic_extent>
+        constexpr span<element_type, see below> subspan() const;
+
+    constexpr span<element_type, dynamic_extent> first(size_type count) const;
+    constexpr span<element_type, dynamic_extent> last(size_type count) const;
+    constexpr span<element_type, dynamic_extent> subspan(size_type offset, size_type count = dynamic_extent) const;
+
+    // [span.obs], span observers
+    constexpr size_type size() const noexcept;
+    constexpr size_type size_bytes() const noexcept;
+    constexpr bool empty() const noexcept;
+
+    // [span.elem], span element access
+    constexpr reference operator[](size_type idx) const;
+    constexpr reference front() const;
+    constexpr reference back() const;
+    constexpr pointer data() const noexcept;
+
+    // [span.iterators], span iterator support
+    constexpr iterator begin() const noexcept;
+    constexpr iterator end() const noexcept;
+    constexpr reverse_iterator rbegin() const noexcept;
+    constexpr reverse_iterator rend() const noexcept;
+
+private:
+    pointer data_;    // exposition only
+    size_type size_;  // exposition only
+};
+
+template<class T, size_t N>
+    span(T (&)[N]) -> span<T, N>;
+
+template<class T, size_t N>
+    span(array<T, N>&) -> span<T, N>;
+
+template<class T, size_t N>
+    span(const array<T, N>&) -> span<const T, N>;
+
+template<class Container>
+    span(Container&) -> span<typename Container::value_type>;
+
+template<class Container>
+    span(const Container&) -> span<const typename Container::value_type>;
+
+} // namespace std
+
+*/
+
+//CP
+//#include <__config>
+#include "libcxx_config.hpp"
+
+//CP - redefine _VSTD
+#undef _VSTD
+#define _VSTD std
+
+#include <array>        // for array
+#include <cstddef>      // for byte
+#include <iterator>     // for iterators
+#include <type_traits>  // for remove_cv, etc
+
+
+
+#if !defined(_LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER)
+#pragma GCC system_header
+#endif
+
+_LIBCPP_PUSH_MACROS
+//CP
+//#include <__undef_macros>
+#include "libcxx_undefmacros.hpp"
+
+//CP adjust namespace declaration
+//_LIBCPP_BEGIN_NAMESPACE_STD
+namespace std {
```

Comment:
Do we really want to put it to `std` namespace?
```suggestion
namespace syclstd {
```

Reply:
@alexbatashev .  Yes, it shouldn't be std.

However, wouldn't 
```__SYCL_INLINE_NAMESPACE(cl) {
namespace sycl {
```
be preferable?

Reply:
@cperkinsintel either way works for me

Reply:
done